Engine/Results/BaseResultsHandler.cs

```diff
@@ -509,6 +524,90 @@ protected virtual void SampleBenchmark(DateTime time, decimal value)
+        /// </summary>
+        /// <param name="time">Time of the sample</param>
+        /// <param name="currentPortfolioValue">Current equity value</param>
+        protected virtual void SampleDrawdown(DateTime time, decimal currentPortfolioValue)
+        {
+            // This will throw otherwise, in this case just don't sample
+            if (CumulativeMaxPortfolioValue != 0)
+            {
+                // Calculate our drawdown and sample it
+                var drawdown = Statistics.Statistics.DrawdownPercent(currentPortfolioValue, CumulativeMaxPortfolioValue);
+                Sample("Drawdown", "Equity Drawdown", 0, SeriesType.Line, time, drawdown, "%");
+            }
+        }
+
+        /// <summary>
+        /// Sample assets sales volume
+        /// </summary>
+        /// <param name="time">Time of the sample</param>
+        protected virtual void SampleSalesVolume(DateTime time)
+        {
+            var volumeBySymbol = Algorithm.Portfolio.Values
+                .Select(x => new KeyValuePair<Symbol, decimal>(x.Symbol, x.TotalSaleVolume));
+
+            foreach (var kvp in volumeBySymbol)
+            {
+                Sample("Assets Sales Volume", $"{kvp.Key.Value}", 0, SeriesType.Treemap, time,
+                    kvp.Value, AlgorithmCurrencySymbol);
+            }
+        }
+
+        /// <summary>
+        /// Sample portfolio exposure
+        /// </summary>
+        /// <param name="time">Time of the sample</param>
+        /// <param name="currentPortfolioValue">Current value of the portfolio</param>
+        protected virtual void SampleExposure(DateTime time, decimal currentPortfolioValue)
+        {
+            // Shorts = holdings < 0 , longs = holdings > 0
+            var shortHoldings = Algorithm.Portfolio.Values.Where(holding => holding.HoldingsValue < 0);
+            var longHoldings = Algorithm.Portfolio.Values.Where(holding => holding.HoldingsValue > 0);
+
+            var shortBySecurityType = SumHoldingsBySecurityType(shortHoldings);
+            foreach (var shortHolding in shortBySecurityType)
+            {
+                Sample("Exposure", $"{shortHolding.Key} - Short", 0, SeriesType.Line, time,
+                    shortHolding.Value/currentPortfolioValue, "");
+            }
+
+            var longBySecurityType = SumHoldingsBySecurityType(longHoldings);
+            foreach (var longHolding in longBySecurityType)
+            {
+                Sample("Exposure", $"{longHolding.Key} - Long", 0, SeriesType.Line, time,
+                    longHolding.Value/currentPortfolioValue, "");
+            }
+        }
+
+        /// <summary>
+        /// Helper method for SampleExposure which separates security holdings by type and then sums their holdings
+        /// </summary>
+        /// <param name="holdings"></param>
+        /// <returns></returns>
+        private static Dictionary<SecurityType, decimal> SumHoldingsBySecurityType(IEnumerable<SecurityHolding> holdings)
```

Comment:
nit - Maybe can replace `SumHoldingsBySecurityType` for a call to `Aggregate`?

Reply:
Opt'd for group by then sum, not sure how to apply aggregate for this case 😁